python/cudf_polars/cudf_polars/experimental/spilling.py

```diff
@@ -65,13 +68,37 @@ def wrap_func_spillable(
         The function to be wrapped.
     make_func_output_spillable
         Whether to wrap the function's output in a SpillableWrapper.
+    target_partition_size
+        Target byte size for IO tasks.
 
     Returns
     -------
     A wrapped function that processes spillable DataFrames.
     """
 
     def wrapper(*args: Any) -> Any:
+        # Make headroom before executing the task
+        try:
+            from dask.sizeof import sizeof
+            from distributed import get_worker
+            from rapidsmpf.integrations.dask.core import get_worker_context
+
+            headroom = 0
+            probable_io_task = True
+            for arg in args:
+                if isinstance(arg, SpillableWrapper) and arg._on_host is not None:
```

Comment:
@madsbk - I'd rather not refer to a private attribute here. Any suggestions?

Reply:
You can use [`SpillableWrapper.mem_type()`](https://github.com/rapidsai/rapidsmpf/blob/132a57baac65a28e0502393b466fe4bd12aec750/python/rapidsmpf/rapidsmpf/integrations/dask/spilling.py#L90)